src/Adapter/MySQL.php

```diff
@@ -321,22 +323,76 @@ private function computeOrderByField(array $filterToTableMapping)
             $orderField = $this->getOrderDirection() === 'asc' ? 'price_min' : 'price_max';
         }
 
-        if (array_key_exists($orderField, $filterToTableMapping)
+        $orderField = $this->computeFieldName($orderField, $filterToTableMapping);
+
+        // put some products at the end of the list
+        $orderField = $this->computeShowLast($orderField);
+
+        return $orderField;
+    }
+
+    /**
+     * Sort product list: InStock, OOPS with qty 0, OutOfStock
+     *
+     * @param string $orderField
+     *
+     * @return string
+     */
+    private function computeShowLast($orderField)
```

Comment:
You should pass `$filterToTableMapping` in parameter to use it on line 354 and 357 for performance :)

Reply:
Thanks 😁 Commit: https://github.com/PrestaShop/ps_facetedsearch/pull/154/commits/d33c52aa261319f1264819136dde403b3245847f